crates/ruff/src/rules/flake8_annotations/settings.rs

```diff
@@ -93,3 +95,17 @@ impl From<Settings> for Options {
         }
     }
 }
+
+impl CombinePluginOptions for Options {
+    fn combine(self, other: Self) -> Self {
+        Self {
+            mypy_init_return: self.mypy_init_return.or(other.mypy_init_return),
+            suppress_dummy_args: self.suppress_dummy_args.or(other.suppress_dummy_args),
+            suppress_none_returning: self
+                .suppress_none_returning
+                .or(other.suppress_none_returning),
+            allow_star_arg_any: self.allow_star_arg_any.or(other.allow_star_arg_any),
+            ignore_fully_untyped: self.ignore_fully_untyped.or(other.ignore_fully_untyped),
```

Comment:
I bet we could code-gen these via a Derive macro (you could look at how `ConfigurationOptions` works as an example), if you're interested.

Reply:
Not done any macro stuff before, but I can give it a go. Would this potentially run into an issue with [`flake8_gettext`](https://github.com/charliermarsh/ruff/pull/4431/files#diff-31f36219ba85fb7b29ddb087a9bf221cdec719f50d5af228a2f6e3dc3de1ce1aR79-R83) since this is using a `Vec` instead of an `Option<T>`? This is the only occurrence of not using `Option<T>` that I came across, so arguably I could just not use a derive for that instance, but it does strike me as slightly strange that this isn't an `Option<T>`. I see no reference/reasoning for the difference given in the associated commits/PR.

Reply:
Is that the only field that's not `Option`?

Reply:
Derive macro ended up not being too difficult to implement once I'd figured out how it worked, so I've handled the `Vec<T>` field case as well. So now all `Options` structs have their `CombinePluginOptions` impls derived (at least the ones that had an impl in the first place).

Reply:
> Is that the only field that's not `Option`?

Yep thats the only field in all the plugin options thats not `Option` that I could find.